### ISSUE TITLE
Integrate JSC audit of parameter and return code checking

### ISSUE THREAD

**Issue by skliper** [nasa/osal#135]
Originally part of trac #45 and split off for CCB review purposes.

JSC has done an audit of input parameter checking and intermediate return code checking and added multiple checks/returns.

Primarily affects vxworks6 but some posix code was touched as well.

**Comment by skliper** [comment]
_Imported from trac issue 112.  Created by jphickey on 2015-11-06T16:44:37, last modified: 2015-12-09T11:12:25_

**Comment by skliper** [comment]
_Trac comment by jphickey on 2015-11-09 10:57:25:_

Commit [changeset:093359f] is available for CCB review, branch `trac-112-add-jsc-retcode-parameter-audits`

I have reviewed this changeset and there are a few possible string buffer overrun issues still present in this code.  Some of these could be pre-existing issues.

* in `OS_mkfs()` and `OS_initfs()` the `local_volname` is sized as `OS_MAX_PATH_LEN` but it is sourced from the passed-in `volname` parameter which is sized checked against `OS_FS_VOL_NAME_LEN`.  In the current config it appears that OS_MAX_PATH_LEN is 64 but OS_FS_VOL_NAME_LEN is 32 so this is OK but this is config-specific.  Also a ":0" is appended to local_volname.  The string is also written to `PhysDevName` which is yet a different length `OS_FS_PHYS_NAME_LEN` and not specifically checked against this length either.

* In `OS_GetPhysDeviceName()` the `tempName` variable is hardcoded at length 32.  Since `snprintf()` is used to fill it, that is OK, but later it is used to file `PhysDevName` //output// buffer which is assumed to be `OS_FS_PHYS_NAME_LEN` in length.  The `tempName` variable should probably be sized to `OS_FS_PHYS_NAME_LEN`, not 32.  Note there may also be some risks to taking out the step of filling the buffer in case of error, in case users of this API are not actually checking the return code (likely).

**Comment by skliper** [comment]
_Trac comment by jphickey on 2015-11-17 12:12:06:_

Note - the commit above is based on the `ic-2015-11-10` branch as it depends on #85 (which is merged into that IC branch).

**Comment by skliper** [comment]
_Trac comment by glimes on 2015-11-20 15:36:24:_

Now part of ic-2015-11-20

**Comment by skliper** [comment]
_Trac comment by abrown4 on 2015-11-25 11:38:44:_

Traceability: This issue's changeset [changeset:093359f] picks up changes from #88.  However, #88 also has some recommended documentation changes.

**Comment by skliper** [comment]
_Trac comment by abrown4 on 2015-11-25 12:36:24:_

Traceability: This issues's changeset [changeset:093359f] picks up changes from #98.  (Note that #98 recommends a future (minor) API change to better report that case.  Suggest fixing or opening another ticket.)  Recommend closing #98 in favor of this issue.

**Comment by skliper** [comment]
_Trac comment by abrown4 on 2015-11-25 13:01:53:_

Traceability: This issue's [changeset:093359f] picks up changes from #100.  Recommend closing #100.
Traceability: This issue's [changeset:093359f] picks up changes from #101.  Recommend closing #101.
Traceability: This issue's [changeset:093359f] picks up changes from #107.  Recommend closing #107.
Traceability: This issue's [changeset:093359f] picks up changes from #109.  Recommend closing #109.
Traceability: This issue's [changeset:093359f] picks up changes from #110.  Recommend closing #110.
Traceability: This issue's [changeset:093359f] picks up changes from #114.  Recommend closing #114.
Traceability: This issue's [changeset:093359f] picks up changes from #115.  Recommend closing #115.
Traceability: This issue's [changeset:093359f] picks up changes from #117.  Recommend closing #117.

**Comment by skliper** [comment]
_Trac comment by glimes on 2015-12-02 10:46:20:_

 ic-2015-11-20 is now included in the development branch.
